processor/src/main/resources/org/mapstruct/ap/internal/model/BeanMappingMethod.ftl

```diff
@@ -24,6 +24,19 @@
     }
     </#if>
 
+    <#if hasSubClassMappings()>
+        <#assign first = true />
+        <#list constructSubClassMappings(parameters) as subClass>
+            <#if !first>else</#if> if (${subClass.sourceArgument} instanceof <@includeModel object=subClass.sourceType/>) {
+                <@includeModel object=subClass.assignment existingInstanceMapping=existingInstanceMapping/>
+            }
+            <#assign first = false />
+        </#list>
```

Comment:
no need to use the assignment for first. Check the [Accessing iteration state](https://freemarker.apache.org/docs/ref_directive_list.html#ref_list_accessing_state) for Freemarker. You can use `<#if subClass_index == 0>` instead. We are using that for the `afterMappingReferences`. I see that Freemarker has also added some more [Loop variable built-ins](https://freemarker.apache.org/docs/ref_builtins_loop_var.html), but we are not on that version yet 😄